src/common/models/states.model.ts

```diff
@@ -1,10 +1,31 @@
-export interface IPaginationState {
-  type: 'infinite-scroll' | 'buttons'
+export type IPaginationState = IQueryPaginationState | IJSONBodyPaginationState;
+
+export const isQueryPaginationState = (obj: IPaginationState): obj is IQueryPaginationState => {
```

Comment:
Is it the same function from `config.model`? Why the duplication?

Reply:
The duplication is here because of the type casting functionality these methods allow for. I can simplify this into a method that just returns a boolean if you want to, but I kinda like these methods as they allow for this pattern:

```
const obj: BaseClass = ...;
if(isClassA(obj)) {
    // obj has type signature of A
}
```

The alternative would be:

```
const obj: BaseClass = ...;
if(isClassAWithoutTypeLevelFunction) {
    // obj has still signature of BaseClass
}
```

Reply:
Hmm how about exporting all of this type checking methods to an external helper file?